### DISCUSSION TITLE
Videos on iphone sometimes doesn't start — in web and android works ok.

### DISCUSSION THREAD

**Discussion by strangesound** [martinlaxenaire/curtainsjs#109] [Q&A]
Here is prototype https://artmonta.netlify.app/
It works norm in web and in android devices (in which i tested)
But in iphone there troubles — sometimes one or two or three videos doesn't start.

If I connect iphone to comp to debug — I see this errors (screen)
What can help? Thanks!

<img width="768" alt="Screenshot 2022-12-21 at 10 49 34" src="https://user-images.githubusercontent.com/42893513/208851833-cec53772-d8d6-4b55-8cfe-a1df27fb68b9.png">
<img width="1288" alt="Screenshot 2022-12-21 at 10 49 25" src="https://user-images.githubusercontent.com/42893513/208851838-c0e91ccb-9e21-4fd7-b487-e0c484ed023e.png">
<img width="805" alt="Screenshot 2022-12-21 at 10 37 13" src="https://user-images.githubusercontent.com/42893513/208851841-00793f70-0e3d-408e-80c4-72100030b83c.png">
<img width="1285" alt="Screenshot 2022-12-21 at 10 36 31" src="https://user-images.githubusercontent.com/42893513/208851844-d0759691-f257-4d30-9cbd-8fd62a7da7a4.png">

**Comment by martinlaxenaire** [comment]
Hi @strangesound,

That's weird. I could add a basic check to ensure `this.source` is defined inside the `_videoFrameCallback` method tho, just to be safe.
Could you please post a minimal codepen / jsfiddle reproducing the issue?

Cheers,

**Reply by strangesound** [reply]
Hi! Thanks for the reply
This is my repository on github. Can it help?

https://github.com/strangesound/videoTransitions.git

**Reply by strangesound** [reply]
I tried to put my code to the codepen, but there is so much problems with download shaders/cors/ and others (

**Reply by strangesound** [reply]
Also tested in desktop safari — same problem sometimes — video doesn't start, black screen.

**Reply by strangesound** [reply]
@martinlaxenaire Martin, do you have an option for the paid support to your library?
